### ISSUE TITLE
SBT throws an exception instead of info message if project override testGrouping and contains no tests.

### ISSUE THREAD

**Issue by ezh** [sbt/sbt#777]
SBT throws an exception instead of info message if project override testGrouping and contains no tests.

sbt 0.12.4-RC1

SBT project with

```
    testGrouping <<= (definedTests in Test) map { tests =>
      tests map { test =>
        new Tests.Group(
          name = test.name,
          tests = Seq(test),
          runPolicy = Tests.SubProcess(javaOptions = Seq.empty[String]))
      }
    }
```

Current behavior:

```
> test
[trace] Stack trace suppressed: run last test:execute-tests for the full output.
[error] (test:execute-tests) java.lang.UnsupportedOperationException: empty.reduceLeft
[error] Total time: 0 s, completed 07.06.2013 17:43:36
> last test:execute-tests
java.lang.UnsupportedOperationException: empty.reduceLeft
        at scala.collection.LinearSeqOptimized$class.reduceLeft(LinearSeqOptimized.scala:124)
        at scala.collection.immutable.List.reduceLeft(List.scala:76)
        at scala.collection.TraversableOnce$class.reduce(TraversableOnce.scala:180)
        at scala.collection.immutable.List.reduce(List.scala:76)
        at sbt.Tests$$anonfun$foldTasks$2.apply(Tests.scala:137)
        at sbt.Tests$$anonfun$foldTasks$2.apply(Tests.scala:135)
        at scala.Function1$$anonfun$compose$1.apply(Function1.scala:49)
        at scala.Function1$$anonfun$compose$1.apply(Function1.scala:49)
        at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:41)
        at sbt.std.Transform$$anon$5.work(System.scala:71)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:232)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:232)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:18)
        at sbt.Execute.work(Execute.scala:238)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:232)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:232)
        at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:160)
        at sbt.CompletionService$$anon$2.call(CompletionService.scala:30)
        at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
        at java.util.concurrent.FutureTask.run(FutureTask.java:138)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:439)
        at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
        at java.util.concurrent.FutureTask.run(FutureTask.java:138)
        at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:886)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:908)
        at java.lang.Thread.run(Thread.java:662)
```

Expected behavior:

```
> test
[info] No tests to run for test:test
[success] Total time: 0 s, completed 07.06.2013 17:37:42
```

**Comment by harrah** [comment]
Thanks for the minimal test case.
- works in 0.13, added test case to resolve there
- not a regression from 0.12.3, so it won't merit RC3
- issue is now fixed for 0.12 and will be included in 0.12.5 if there isn't a 0.12.4-RC3

**Comment by sandreev** [comment]
Hi,

I can reproduce the issue with sbt 0.13.1 if I specify the testGrouping in Test:

testGrouping in Test <<= ....

I have checked the relevant lines in Tests.scala but was not able to find them in the foldTasks method.

Might be a regression, could you please backport the fix to sbt 0.13?
